added error cases for 400 query timeout

jrchudy opened the pull request:
_body pending_

jrchudy commented:
@RFSH  Some changes that I made to accommodate the simplified facet mode:
 - `retryQuery` and `resetFacet` functions for options in facet.
    - should `hasError` be set to `false` here? it is only used for the QTE case and avoids throwing the error
    - I'm thinking it might need to be done "only" in success callbacks (regardless of it's previous value)
 - updateCount bug
    - if [this](https://github.com/informatics-isi-edu/chaise/blob/master/common/table.js#L377-L385) catch clause is reached and it make it to the "fail silently" comment, no value is returned to the calling function and therefore `_afterUpdateCount` is not called. This causes the flow control `occupiedSlots` count to be incorrect and will eventually cause all requests to "hang"
 - Should the changes apply to range facets as well?

RFSH commented on common/faceting.js:
```diff
@@ -72,6 +73,10 @@
 
+                    /**
+                     * @param {int} index index of facetColumn
+                     * @param {boolean} noConstraints set property to run the query without constraints
```
It seems like you've removed this parameter

jrchudy commented on common/table.js:
```diff
@@ -380,8 +402,15 @@
                     }
 
+                    if (err instanceof ERMrest.QueryTimeoutError) {
+                        vm.countError = true;
```
Right, this should be done in the success callbacks. I'm going to change the `hasError` properties so that the errors are `facetError`, `tableError`, `countError`, `tableError`.

jrchudy commented:
400 Query Timeout in facet:
   - Simplify button tooltip -> "Retry the query that timed out without any other facet constraints applied."
![image](https://user-images.githubusercontent.com/2932600/46373635-61293880-c643-11e8-95f0-07106158ee66.png)

After clicking "Simplify" from above picture:
  - Reset link tooltip -> "Reset"
![image](https://user-images.githubusercontent.com/2932600/46373895-27a4fd00-c644-11e8-9e59-263e8f6e976c.png)

jrchudy commented:
@RFSH  I updated this issue with the changes to the text that @hongsudt suggested. I also added a class to each ellipses cell to change the background color of it if the current aggregate column has an error.

jrchudy commented on common/templates/ellipses.html:
```diff
@@ -21,7 +21,7 @@
     </div>
 </td>
-<td ng-if="config.showNull || tuple.uniqueId" ng-repeat="val in rowValues track by $index">
+<td ng-if="config.showNull || tuple.uniqueId" ng-repeat="val in rowValues track by $index" ng-class="{'disabled-cell': columnModels[$index].columnError}">
```
This is why the ellipses directive (our row directive) has all the columnModels attached to it.

RFSH commented on common/faceting.js:
```diff
@@ -59,6 +59,7 @@
                         ctrl.facetingCount++;
 
                         $scope.vm.facetModels[index] = {
+                            facetError: false,
```
Just for the sake of bookkeeping, can you please add the `noConstraints:false` to here too. So we would know that the facetModel can have such attribute.
